### PR TITLE
Add nyc and codecov to TravisCI

### PR DESCRIPTION
**Pull request by coditva**
TODO:
- [x] update coverage targets for tests in `codecov.yml`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,13 +23,13 @@
...
   "scripts": {
-    "test": "node npm/test.js",
+    "test": "npm run test-lint && npm run test-system && npm run test-unit && npm run test-integration && npm run test-cli && npm run test-library",
     "test-system": "node npm/test-system.js",
     "test-lint": "node npm/test-lint.js",
     "test-unit": "node npm/test-unit.js",
-    "test-integration": "node npm/test-integration.js",
-    "test-cli": "node npm/test-cli.js",
-    "test-library": "node npm/test-library.js",
+    "test-integration": "nyc --temp-dir .coverage --silent node npm/test-integration.js",
```
Comment (coditva) [inline package.json]: `nyc` reports coverage as "unknown" unless the whole file is executed with nyc as done here. @codenirvana

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        target: 40
+
+      # coverage status for integration tests
+      integration:
+        flags: integration
+        target: 70
+
+      # coverage status for cli tests
+      integration:
+        flags: integration
```
Comment (codenirvana) [inline codecov.yml]: `integration` or `cli`?

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        target: 70
+
+      # coverage status for library tests
+      integration:
+        flags: integration
```
Comment (codenirvana) [inline codecov.yml]: `integration` or `library`?

**Comment by coditva** [comment]
No change but forced pushed commit to restart Travis. Required for merging.

**Pull request merged**